web/client/components/background/BackgroundSelector.jsx

```diff
@@ -6,17 +6,19 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React from 'react';
+import React, { lazy } from 'react';
 
 import assign from 'object-assign';
 import Message from '../I18N/Message';
 import PreviewButton from './PreviewButton';
 import PreviewList from './PreviewList';
 import PreviewIcon from './PreviewIcon';
 import ToolbarButton from '../misc/toolbar/ToolbarButton';
-import BackgroundDialog from './BackgroundDialog';
+
 import ConfirmDialog from '../misc/ConfirmDialog';
 import PropTypes from 'prop-types';
+import withSuspense from '../misc/withSuspense';
+const BackgroundDialog = withSuspense()(lazy(() => import('./BackgroundDialog')));
```

Comment:
should we document this new approach somewhere? maybe in the training?

Reply:
I agree that we should add a documentation to explain how to create a plugin taking into account the new store api and the lazy approach.
In this issue there is an item on the list to cover this request https://github.com/geosolutions-it/MapStore2/issues/6735#issuecomment-1233077463

Reply:
ok good